bots/foundation/anthropic_bots.py

```
@@ -251,92 +266,168 @@ def __init__(self, verbose: bool = False):
+                    )
+                    if span:
+                        span.add_event(
+                            "api.error.retry", {"attempt": attempt + 1, "error_type": e.__class__.__name__, "delay": delay}
+                        )
+                    time.sleep(delay)
+
+            # If we get here, max retries reached
+            logger.error("Max retries reached, unable to send message", extra={"provider": "anthropic"})
+            if span:
+                span.add_event("max_retries_reached")
+            raise Exception("Max retries reached. Unable to send message.")
```

Ruff (0.13.3)

Create your own exception

(TRY002)